miden-lib/asm/scripts/ASWAP.masm

```diff
@@ -0,0 +1,52 @@
+use.miden::sat::note
+use.miden::wallets::basic->wallet
+
+# Atomic Swap script: adds an asset from the note into consumers account and
+# creates a note consumable by note issuer containing requested ASSET.
+#
+# Requires that the account exposes: 
+#
+# Inputs: [ASSET, serial_num]
+# Outputs: []
+#
+# Note inputs are assumed to be as follows:
+# - asset:
+# - serial_num
+#
+# FAILS if:
+# - Account does not expose miden::wallets::basic::receive_asset procedure
+# - Account does not expose miden::wallets::basic::send_asset procedure
+# - Account vault does not contain the requested asset
+# - Adding a fungible asset would result in amount overflow, i.e., the total amount would be
+#   greater than 2^63
+begin
+    # drop the transaction script root
+    dropw
+    # => []
+
+    # store asset into memory
+    push.3 exec.note::get_assets assert
+    # => [ptr]
```

Comment:
minor nit: I'd clarify the comment to say something like "store asset into memory at address 3".